data_dictionary/data_dictionary.cc

```diff
@@ -365,7 +365,7 @@ storage_options storage_options::append_to_s3_prefix(const sstring& s) const {
     s3 s3_options = std::get<s3>(value);
     SCYLLA_ASSERT(std::holds_alternative<sstring>(s3_options.location));
     sstring prefix = std::get<sstring>(s3_options.location);
-    s3_options.location = prefix + s;
+    s3_options.location = seastar::format("{}/{}", prefix, s);
```

Comment:
```suggestion
    if (!s.empty()) {
        s3_options.location = seastar::format("{}/{}", prefix, s);
    }
```

This works with both the old and the new test.